MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -28,19 +39,57 @@ final class MovieQuizViewController: UIViewController, QuestionFactoryDelegate {
     // MARK: - Lifecycle
     override func viewDidLoad() {
         super.viewDidLoad()
-        questionFactory.delegate = self
-        alertPresenter.controller = self
         prepareMovieImage()
-        questionFactory.requestNextQuestion()
-        showQuestionOrResults()
+        questionFactory.loadData()
     }
     
-    // MARK: - Delegates
+    // MARK: - Delegate methods
     func didReceiveNextQuestion(question: QuizQuestion?) {
         currentQuestion = question
+        DispatchQueue.main.async { [weak self] in
+            self?.showQuestionOrResults()
+        }
+    }
+    
+    func didLoadDataFromServer() {
+        questionFactory.requestNextQuestion()
+    }
+    
+    func didFailToLoadData(with error: Error) {
+        DispatchQueue.main.async { [weak self] in
+            self?.showNetworkError(message: error.localizedDescription)
+        }
+    }
+    
+    func isLoadingInProgress(state: Bool) {
+        DispatchQueue.main.async { [weak self] in
+            guard let self = self else { return }
+            self.activityBackground.isHidden = !state
+            self.setButtonsEnabled(!state)
+            self.activityBackground.alpha = 0
+            UIView.animate(withDuration: 0, delay: 1) {
```

Comment:
тут не поняла, почему именно так захотел сделать, очень странно видеть анимацию с длительностью - 0 :)

В качестве отложенного вызова есть DispatchQueue.main.asyncAfter(deadline: .now() + 1). Будет работать точно также. 

Если хотелось прям анимацию, то нужно давать время на анимацию, например так:
```suggestion
            UIView.animate(withDuration: 0.3, delay: 0.7) {
```
Если цель не показывать лоадер сразу во время загрузки, а только если данные слишком долго грузятся, то так работать будет немного плавнее

Reply:
Тут меня напрягало появление индикатора загрузки на очень короткое время при загрузке картинок. Нашел такой вариант. Смысл в том, что индикатор появляется, но невидимым и если загрузка идет больше 1 секунды становится видимым. А если меньше секунды, то вообще не появится.
В общем меня такой вариант полностью устроил🙂

Reply:
Хорошо :) интересно, хоть и немного не по ТЗ. принимаю)